CPDShell/Core/algorithms/DensityBasedCPD/abstracts/density_based_algorithm.py

```diff
@@ -0,0 +1,30 @@
+import numpy as np
+from scipy.optimize import minimize
+from sklearn.neighbors import KernelDensity
```

Comment:
Did you add this into poetry.lock file? If not, run `poetry add sklearn` in console, so we could install all needed dependencies right from it.